common-service-core/src/main/java/com/ca/mfaas/security/HttpsFactory.java

```diff
@@ -282,47 +251,8 @@ public EurekaJerseyClientBuilder createEurekaJerseyClientBuilder(String eurekaSe
         return builder;
     }
 
+
     public String readSecret() {
-        if (config.getKeyStore() == null) {
-            return null;
-        }
-        try {
-            KeyStore ks = KeyStore.getInstance(config.getKeyStoreType());
-            InputStream istream;
-            if (config.getKeyStore().startsWith(SAFKEYRING)) {
-                URL url = keyRingUrl(config.getKeyStore());
-                istream = url.openStream();
-            } else {
-                File keyStoreFile = new File(config.getKeyStore());
-                istream = new FileInputStream(keyStoreFile);
-            }
-            ks.load(istream, config.getKeyStorePassword() == null ? null : config.getKeyStorePassword().toCharArray());
-            char[] keyPassword = config.getKeyPassword() == null ? null : config.getKeyPassword().toCharArray();
-            Key key = null;
-            if (config.getKeyAlias() != null) {
-                key = ks.getKey(config.getKeyAlias(), keyPassword);
-            } else {
-                for (Enumeration<String> e = ks.aliases(); e.hasMoreElements();) {
-                    String alias = e.nextElement();
-                    try {
-                        key = ks.getKey(alias, keyPassword);
-                        break;
-                    } catch (UnrecoverableKeyException uke) {
-                        log.debug("Key with alias {} could not be used: {}", alias, uke.getMessage());
-                    }
-                }
-            }
-            if (key == null) {
-                throw new UnrecoverableKeyException(String.format(
-                        "No key with private key entry could be used in the keystore. Provided key alias: %s",
-                        config.getKeyAlias() == null ? "<not provided>" : config.getKeyAlias()));
-            }
-            return Base64.getEncoder().encodeToString(key.getEncoded());
-        } catch (NoSuchAlgorithmException | KeyStoreException | CertificateException | IOException
-                | UnrecoverableKeyException e) {
-            log.error("Error reading secret key: {}", e.getMessage(), e);
-            throw new HttpsConfigError("Error reading secret key: " + e.getMessage(), e,
-                    ErrorCode.HTTP_CLIENT_INITIALIZATION_FAILED, config);
-        }
+        return SecurityUtils.readSecret(config);
```

Comment:
I couldn't see where it's used. Maybe someone can show me where. I found in `HttpConfig.class` it's called but `getSecret` is not called. Looks useless.
```
 public String getSecret() {
        return secret;
    }
```

Reply:
`readSecret()` used only in `HttpsFactoryTest`, so, you are right it's redundant the same as `getSecret()` in `HttpConfig`